sklearn/utils/estimator_checks.py

```diff
@@ -687,7 +691,11 @@ def check_fit_score_takes_y(name, Estimator):
 @ignore_warnings
 def check_estimators_dtypes(name, Estimator):
     rnd = np.random.RandomState(0)
-    X_train_32 = 3 * rnd.uniform(size=(20, 5)).astype(np.float32)
+    if name in ["BoxCoxTransformer", "SpectralCoclustering",
```

Comment:
Even I am not familiar at all with the algorithm, but when the current code is executed, ``normalized_data`` [here](https://github.com/maniteja123/scikit-learn/blob/bf459f445c43a663df565bc0aa237f57acfc7264/sklearn/cluster/bicluster.py#L293) has ``NaN`` and fails finite check during [this](https://github.com/maniteja123/scikit-learn/blob/bf459f445c43a663df565bc0aa237f57acfc7264/sklearn/cluster/bicluster.py#L147) call.